src/main/java/org/opensearch/ad/rest/RestIndexAnomalyDetectorAction.java

```diff
@@ -97,57 +141,70 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
 
-    @Override
-    public List<ReplacedRoute> replacedRoutes() {
-        return ImmutableList
-            .of(
-                // Create
-                new ReplacedRoute(
-                    RestRequest.Method.POST,
-                    AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI,
-                    RestRequest.Method.POST,
-                    AnomalyDetectorPlugin.LEGACY_OPENDISTRO_AD_BASE_URI
-                ),
-                // Update
-                new ReplacedRoute(
-                    RestRequest.Method.PUT,
-                    String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, DETECTOR_ID),
-                    RestRequest.Method.PUT,
-                    String.format(Locale.ROOT, "%s/{%s}", AnomalyDetectorPlugin.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID)
-                )
+        CompletableFuture<IndexAnomalyDetectorResponse> futureResponse = new CompletableFuture<>();
+        indexAction
+            .doExecute(
+                null,
+                indexAnomalyDetectorRequest,
+                ActionListener.wrap(r -> futureResponse.complete(r), e -> futureResponse.completeExceptionally(e))
             );
+
+        IndexAnomalyDetectorResponse response = futureResponse
+            .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(environmentSettings).getMillis(), TimeUnit.MILLISECONDS)
+            .join();
+        // TODO handle exceptional response
+        return indexAnomalyDetectorResponse(request, response);
     }
 
-    private RestResponseListener<IndexAnomalyDetectorResponse> indexAnomalyDetectorResponse(
-        RestChannel channel,
-        RestRequest.Method method
-    ) {
-        return new RestResponseListener<IndexAnomalyDetectorResponse>(channel) {
-            @Override
-            public RestResponse buildResponse(IndexAnomalyDetectorResponse response) throws Exception {
-                RestStatus restStatus = RestStatus.CREATED;
-                if (method == RestRequest.Method.PUT) {
-                    restStatus = RestStatus.OK;
-                }
-                BytesRestResponse bytesRestResponse = new BytesRestResponse(
-                    restStatus,
-                    response.toXContent(channel.newBuilder(), ToXContent.EMPTY_PARAMS)
-                );
-                if (restStatus == RestStatus.CREATED) {
-                    String location = String.format(Locale.ROOT, "%s/%s", AnomalyDetectorPlugin.LEGACY_AD_BASE, response.getId());
-                    bytesRestResponse.addHeader("Location", location);
-                }
-                return bytesRestResponse;
-            }
-        };
+    private ExtensionRestResponse indexAnomalyDetectorResponse(ExtensionRestRequest request, IndexAnomalyDetectorResponse response)
+        throws IOException {
+        RestStatus restStatus = RestStatus.CREATED;
+        if (request.method() == RestRequest.Method.PUT) {
+            restStatus = RestStatus.OK;
+        } else {
```

Comment:
Is this else just for debugging purpose?

Reply:
Yes and no.  Early on I was getting timeouts on my OpenSearch rest requests, but it was successfully creating the detector, but I didn't have the ID.   But it seems useful to add the detector ID to the logs for the CREATED case.   So I added it for debugging but think it's useful to keep.